File: src/layout/layoutDetectionEngine.ts
/**
 * Converted from Python: core/layout/layout_detection_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export let MAX_BLOCKS: any = 10000;
export function detectLayoutBlocks(ocr_regions: any): any {
  var blocks: any[] = [];
  var idx: any;
  var region: any;
  for ([idx, region] of py.enumerate(py.slice(ocr_regions, null, MAX_BLOCKS))) {
    py.listAppend(blocks, {"id": `block_${py.toStr(idx)}`, "bbox": py.get(region, "bbox"), "text": py.get(region, "text", ""), "type": "text_block"});
  }
  return {"blocks": blocks, "bounded": true};
}

File: lib/src/layout/layout_detection_engine.dart
/// Port of core/layout/layout_detection_engine.py: detect_layout_blocks.
library;

const int maxBlocks = 10000;

/// Port of core/layout/layout_detection_engine.detect_layout_blocks.
Map<String, dynamic> detectLayoutBlocks(
  List<Map<String, dynamic>> ocrRegions,
) {
  final List<Map<String, dynamic>> blocks = <Map<String, dynamic>>[];

  final int limit =
      ocrRegions.length > maxBlocks ? maxBlocks : ocrRegions.length;

  for (int idx = 0; idx < limit; idx++) {
    final Map<String, dynamic> region = ocrRegions[idx];
    blocks.add(<String, dynamic>{
      'id': 'block_$idx',
      'bbox': region['bbox'],
      'text': region.containsKey('text') ? region['text'] : '',
      'type': 'text_block',
    });
  }

  return <String, dynamic>{
    'blocks': blocks,
    'bounded': true,
  };
}
